tests/integration/solidity_contracts/Counter/test_counter.py

```diff
@@ -226,3 +226,166 @@ async def test_extcodecopy_offset_and_size_gte_twenty_c(
             memory_result[dest_offset : size + dest_offset]
             == expected_memory_result[offset : offset + size]
         )
+
+    @pytest.mark.parametrize(
+        "case",
+        [
+            {
+                "bytecode_match": {
+                    "offset": 0,
+                    "size": 32,
+                    "dest_offset": 0,
+                },
+                "bytecode_zeroed": {
+                    "offset": 0,
+                    "size": 8,
+                    "dest_offset": 1,
+                    "contract_address": "mock",
+                },
+                "id": "no account match",
+            },
+            {
+                "bytecode_match": {
+                    "offset": 0,
+                    "size": 32,
+                    "dest_offset": 0,
+                },
+                "bytecode_zeroed": {
+                    "offset": 0,
+                    "size": 8,
+                    "dest_offset": 1,
+                    "contract_address": "counter",
+                },
+                "id": "offset exceeds match length",
+            },
+        ],
+    )
+    async def test_extcodecopy_should_pad_zeroes(
```

Comment:
there's something I'm missing with this test. I don't understand the need for to`bytecode` per case. From my understanding, there are 3 cases:
- requested code is shorter than existing
- requested code is partly existing partly padded
- requested code is out of range and returns only 0

Then the test could be

```
counter = await deploy_solidity_contract("Counter", caller_address=addresses[1]["int"])
bytecode_len, bytecode = (await counter.contract_account.bytecode()).result

# instructions to EXTCODECOPY with offset/size depending on case; relative to `bytecode_len`
offset =
size = 
expected_bytecode = bytecode[offset: (offset + size)]

res = await kakarot.execute(
            value=int(0),
            bytecode=hex_string_to_bytes_array(instructions),
            calldata=hex_string_to_bytes_array(""),
        ).call()

memory_result = extract_memory_from_execute(res.result)
assert memory[dest_offset: (dest_offset + size)] == expected_bytecode
```

what am I missing?

Reply:
If I am following what you mean by `bytecode` per case:

There is also the case that the account for the requested code has no code associated, right? And to verify that the opcode is handling this case correctly (as opposed to just returning nothing, as was happening before) the test was constructed to ensure that the zeroes contrast with bytecode in memory. Does that make sense?

Reply:
just wanted to document here your helping me understand the difference between

- `counter.bytecode` this is bytecode method for web3.Contract
- `counter.contract_account.bytecode` is bytecode method of contract_account.cairo

thanks again!

Reply:
Are we good-to-go?